### DISCUSSION TITLE
Update user data in `authenticator.isAuthenticated(request)`

### DISCUSSION THREAD

**Discussion by kavinvalli** [sergiodxa/remix-auth#136] [Q&A] [answered]
I'm currently using this to get the user
```ts
  const user = await authenticator.isAuthenticated(request);
```

Now, if I'm making a change to the user, it's not updated in the session user data. Is there some way to do that?

**Comment by sergiodxa** [comment, marked as answer]
If you update the user data you need to get the session object, change the `authenticator.sessionKey` with the new value, commit the session and send the new cookie to the browser, on the next request isAuthenticated will give you the need value from the session.

**Reply by kavinvalli** [reply]
Hi. Can you give an example please cause I'm not able to implement this

**Reply by sergiodxa** [reply]
```ts
// get the session object
let session = await sessionStorage.getSession(request.headers.get("Cookie"))

// replace the user data on the session
session.set(authenticator.sessionKey, newUserData)

// commit the session to a cookie string
let cookie = await sessionStorage.commitSession(session)

// and return the cookie in a response, json or redirect for example
return redirect(somePath, { headers: { "Set-Cookie": cookie } })
return json(someData, { headers: { "Set-Cookie": cookie } })
```

**Reply by andreujuanc** [reply]
Would be nice if this was "abstracted away". So we can do just auth.updateValue("key", "value").
